src/main/java/com/fasterxml/jackson/module/paramnames/ParameterNamesAnnotationIntrospector.java

```diff
@@ -14,15 +15,16 @@
  * @see AnnotationIntrospector
  * @see Parameter
  */
-class ParameterNamesAnnotationIntrospector extends NopAnnotationIntrospector
-{
+class ParameterNamesAnnotationIntrospector extends NopAnnotationIntrospector {
     private static final long serialVersionUID = 1L;
 
     private final JsonCreator.Mode creatorBinding;
+    private final ParameterExtractor parameterExtractor;
 
-    ParameterNamesAnnotationIntrospector(JsonCreator.Mode creatorBinding) {
+    ParameterNamesAnnotationIntrospector(JsonCreator.Mode creatorBinding, ParameterExtractor parameterExtractor) {
```

Comment:
ParameterNamesAnnotationIntrospector isn't part of the module public API since the class is package private. Users that do override this class do so, from the issues reported so far, only for a hot fix, so I'm against keeping the old constructor.